lib/spack/spack/build_systems/cmake.py

```diff
@@ -73,14 +68,12 @@ class CMakePackage(PackageBase):
 
     build_time_test_callbacks = ['check']
 
-    depends_on('cmake', type='build')
-
-    def build_type(self):
-        """Returns the correct value for the ``CMAKE_BUILD_TYPE`` variable
+    # https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html
+    variant('build_type', default='RelWithDebInfo',
+            description='The build type to build',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
```

Comment:
We could do that, but then it becomes a lot of work to translate the name from lowercase to CamelCase. Unless CMake doesn't care about capitalization? Even then, I think it looks better as CamelCase and would be closer to what users are expecting.